src/components/CountDown/CountDown.tsx

```diff
@@ -0,0 +1,181 @@
+    const timeout = setTimeout(() => {
+      if (seconds >= 0) {
+        const uff = timeEnd.current - new Date().getTime()
+        setSeconds(uff / milisecond)
+      }
+      if (seconds < 1) {
+        if (loop) {
+          timeEnd.current = timeEnd.current - new Date().getTime()
+          setSeconds(initialSeconds)
+        } else {
+          clearTimeout(timeout)
+          if (onFinish) {
+            onFinish()
+          }
+        }
+      }
+    }, timeLoop)
+    return () => {
+      clearTimeout(timeout)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [seconds, appStateVisible])
+
+  const renderColons = () => (
+    <Container>
+      <Text style={[textStyle, colonsStyle]}>:</Text>
+    </Container>
+  )
+
+  const renderTimer = () => {
+    const numberSecondPerMinute = 60
+    /*
+    caculate minutes
+    */
+    const resultCaculatMinute = Math.floor(seconds / numberSecondPerMinute) % numberSecondPerMinute
+    const minute = resultCaculatMinute >= 0 ? resultCaculatMinute : 0
+    const textMinute = (
+      <Container style={elementStyle}>
+        <Text style={[textStyle]}>{minute >= 10 ? `${minute}` : `0${minute}`}</Text>
+      </Container>
+    )
+    /*
+    caculate seconds
+    */
+    const resultCaculatSecond = Math.round(seconds % numberSecondPerMinute)
+    const second = resultCaculatSecond >= 0 ? resultCaculatSecond : 0
+    const textSecond = (
+      <Container style={elementStyle}>
+        <Text style={[textStyle]}>{second >= 10 ? `${second}` : `0${second}`}</Text>
+      </Container>
+    )
+    let textDay = null
+    let textHour = null
+
+    /*
+    caculate day
+    */
+    if (FormatTime.ddhhmmss === FormatTime[format]) {
+      const secondPerDay = 86400
+      const resultCaculatDay = Math.floor(seconds / secondPerDay)
+      const day = resultCaculatDay >= 0 ? resultCaculatDay : 0
+      textDay = (
+        <Container style={elementStyle}>
+          <Text style={[textStyle]}>{day >= 10 ? `${day}` : `0${day}`}</Text>
+        </Container>
+      )
+    }
+    /*
+    caculate hours
+    */
+    if (FormatTime.hhmmss === FormatTime[format] || FormatTime.ddhhmmss === FormatTime[format]) {
+      const secondPerHour = 3600
+      const resultCaculatHour = Math.floor(seconds / secondPerHour)
+      const hourPerDay = 24
+      const hour = resultCaculatHour >= 0 ? resultCaculatHour % hourPerDay : 0
+      textHour = (
+        <Container style={elementStyle}>
+          <Text style={[textStyle]}>{hour >= 10 ? `${hour}` : `0${hour}`}</Text>
+        </Container>
+      )
+    }
+
+    return (
+      <Container style={containerStyle}>
+        {textDay && textDay}
+        {textDay && renderColons()}
+        {textHour && textHour}
+        {textHour && renderColons()}
+        {textMinute}
+        {renderColons()}
+        {textSecond}
+      </Container>
+    )
+  }
+
+  return renderTimer()
+}
+
+const Container = styled.View({
+  flexDirection: 'row',
+})
```

Comment:
create a const TextBase = styled.Text<TextProps>({theme})...

view some different component to make right